File: LC 2788.cpp
class Solution {
public:
    vector<string> splitWordsBySeparator(vector<string>& words, char separator) {
        vector<string> ans;
        for (auto& s : words) {
            string t;
            for (auto& c : s) {
                if (c == separator) {
                    if (t != "") {
                        ans.push_back(t);
                        t = "";
                    }
                } else {
                    t += c;
                }
            }
            if (t != "") {
                ans.push_back(t);
            }
        }
        return ans;
    }
};

File: 3000/LC 2788.go
func splitWordsBySeparator(words []string, separator byte) (ans []string) {
    for _, s := range words {
        for _, str := range strings.Split(s, string(separator)) {
            if len(str) > 0 {
                ans = append(ans, str)
            }
        }
    }

    return 
}